dotCMS/src/integration-test/java/com/dotmarketing/business/IdentifierFactoryTest.java

```diff
@@ -0,0 +1,550 @@
+    }
+
+    @Test
+    public void testLoadFromDbFound() throws DotDataException {
+        final Identifier identifier = factory.loadFromDb(defaultHost.getIdentifier());
+
+        Assert.assertNotNull(identifier);
+        Assert.assertEquals(defaultHost.getIdentifier(), identifier.getId());
+
+    }
+
+    @Test(expected = DotStateException.class)
+    public void testLoadFromDbWithNullID() throws DotDataException {
+        factory.loadFromDb((String) null);
+    }
+
+    @Test
+    public void testLoadFromDbVersionableFound() throws DotDataException {
+        final Identifier identifier = factory.loadFromDb(defaultHost);
+
+        Assert.assertNotNull(identifier);
+        Assert.assertEquals(defaultHost.getIdentifier(), identifier.getId());
+
+    }
+
+    @Test(expected = DotStateException.class)
+    public void testLoadFromDbWithNullVersionable() throws DotDataException {
+        factory.loadFromDb((Versionable) null);
+    }
+
+    @Test
+    public void testCreateNewFolderIdentifierForFolder()
+            throws DotSecurityException, DotDataException {
+
+        Folder newFolder, parentFolder;
+        Identifier identifier;
+
+        parentFolder = folderAPI.findFolderByPath("/products", defaultHost, systemUser, false);
+        newFolder = new Folder();
+        identifier = null;
+
+        newFolder.setName("TestingFolder" + System.currentTimeMillis());
+
+        try {
+            //Creates new identifier
+            identifier = factory.createNewIdentifier(newFolder, parentFolder);
+
+            Assert.assertNotNull(identifier.getId());
+            Assert.assertFalse(identifier.getId().isEmpty());
+            Assert.assertNotNull(newFolder.getVersionId());
+            Assert.assertFalse(newFolder.getVersionId().isEmpty());
+            Assert.assertEquals(parentFolder.getPath(), identifier.getParentPath());
+            Assert.assertEquals(Identifier.ASSET_TYPE_FOLDER, identifier.getAssetType());
+            Assert.assertEquals(newFolder.getName().toLowerCase(), identifier.getAssetName());
+        } finally {
+            if (identifier != null) {
+                //Deletes the created identifier
+                deleteIdentifier(identifier);
+            }
+        }
+    }
+
+    @Test
+    public void testCreateNewFileAssetIdentifierForFolder()
+            throws DotSecurityException, DotDataException, IOException {
+
+        Contentlet newContentlet;
+        Identifier identifier;
+        File tempFile;
+        Folder parentFolder;
+        String fileName;
+
+        identifier = null;
+        newContentlet = new FileAsset();
+        fileName = "tempFile" + System.currentTimeMillis();
+        tempFile = File.createTempFile(fileName, "txt");
+        parentFolder = folderAPI.findFolderByPath("/products", defaultHost, systemUser, false);
+
+        newContentlet.setContentTypeId(CacheLocator.getContentTypeCache()
+                .getStructureByVelocityVarName(FileAssetAPI.BINARY_FIELD).getInode());
```

Comment:
@nollymar we have (but was hidden because of javadoc) a method that takes velVarName for searching contenttypes in the API. See contentTypeAPI.find(inodeOrVar). Please use that one instead